### PR TITLE
Reverse the control flow for `isolatedProjects` executor enablement 

### LINKED ISSUE #30207: Flip a way `isolatedProjects` test executor being enabled for subprojects

### PR DESCRIPTION
**Pull request by 6hundreds**
Fixes #30207 

Previous opt-in approach had a drawback - when a new subproject added, `isolatedProjectsIntegTest` is being silently disabled for it. 

This PR is flipping it - now subprojects that not supporting IP tests are explicitly listed.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt [hunk 1]
```diff
@@ -173,13 +173,17 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
...
+        disableIfNeeded(project.name, testType, executer)
     }
 
 
 internal
-fun IntegrationTest.disableIfNeeded(testType: TestType, executer: String) {
-    if (testType == TestType.INTEGRATION && executer == "isolatedProjects") {
+fun IntegrationTest.disableIfNeeded(projectName: String, testType: TestType, executer: String) {
+    if (
+        ignoreWithIsolatedProjectsExecuter.contains(projectName) &&
```
Comment (bamboo) [inline build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt:183]: 🚀 Reorder `&&` operands from least expensive to most expensive

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt [hunk 2]
```diff
@@ -276,3 +280,181 @@ fun Project.resolver(name: String, libraryElements: String, extends: Configurati
         extendsFrom(extends)
     }
 }
+
+private val ignoreWithIsolatedProjectsExecuter: List<String> = listOf(
```
Comment (bamboo) [inline build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt:284]: 💅 Explicit type is unnecessary (member is private and it's initialized immediately from a list of strings)

**Review by bamboo — CHANGES_REQUESTED** [review]

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt [hunk 3]
```diff
@@ -276,3 +280,181 @@ fun Project.resolver(name: String, libraryElements: String, extends: Configurati
         extendsFrom(extends)
     }
 }
+
+private val ignoreWithIsolatedProjectsExecuter = listOf(
```
Comment (cobexer) [inline build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt:284]: ❌ please sort this list alphabetically (your IDE can do that for you)

**Review by cobexer — CHANGES_REQUESTED** [review]

File: build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt (hunk 3 above)
Comment (6hundreds) [reply build-logic/integration-testing/src/main/kotlin/gradlebuild/integrationtests/shared-configuration.kt:284]: Good note, did it

**Comment by 6hundreds** [comment]
@cobexer Are we good to go with it?

**Comment by 6hundreds** [comment]
@cobexer I'll merge it, if you have any other concerns - just follow up in this PR

**Pull request merged**